diff -rupN ./DPDKORG/lib/librte_cmdline/Makefile ./DPDK/lib/librte_cmdline/Makefile
--- ./DPDKORG/lib/librte_cmdline/Makefile	2013-08-02 11:28:34.490127868 +1200
+++ ./DPDK/lib/librte_cmdline/Makefile	2013-07-31 17:05:14.386374791 +1200
@@ -31,7 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
-
+include $(RTE_SDK)/mk/rte.shared.mk
 # library name
 LIB = librte_cmdline.a
 
diff -rupN ./DPDKORG/lib/librte_eal/linuxapp/eal/Makefile ./DPDK/lib/librte_eal/linuxapp/eal/Makefile
--- ./DPDKORG/lib/librte_eal/linuxapp/eal/Makefile	2013-08-02 11:28:34.478374230 +1200
+++ ./DPDK/lib/librte_eal/linuxapp/eal/Makefile	2013-07-31 17:03:12.650385814 +1200
@@ -31,7 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
-
+include $(RTE_SDK)/mk/rte.shared.mk
 LIB = librte_eal.a
 
 VPATH += $(RTE_SDK)/lib/librte_eal/common
diff -rupN ./DPDKORG/lib/librte_ether/Makefile ./DPDK/lib/librte_ether/Makefile
--- ./DPDKORG/lib/librte_ether/Makefile	2013-08-02 11:28:34.490127868 +1200
+++ ./DPDK/lib/librte_ether/Makefile	2013-07-31 17:06:01.394168270 +1200
@@ -31,7 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
-
+include $(RTE_SDK)/mk/rte.shared.mk
 #
 # library name
 #
diff -rupN ./DPDKORG/lib/librte_hash/Makefile ./DPDK/lib/librte_hash/Makefile
--- ./DPDKORG/lib/librte_hash/Makefile	2013-08-02 11:28:34.486374904 +1200
+++ ./DPDK/lib/librte_hash/Makefile	2013-07-31 17:07:00.034284363 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_hash.a
diff -rupN ./DPDKORG/lib/librte_kni/Makefile ./DPDK/lib/librte_kni/Makefile
--- ./DPDKORG/lib/librte_kni/Makefile	2013-08-02 11:28:34.490127868 +1200
+++ ./DPDK/lib/librte_kni/Makefile	2013-07-31 17:08:06.231125098 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_kni.a
diff -rupN ./DPDKORG/lib/librte_lpm/Makefile ./DPDK/lib/librte_lpm/Makefile
--- ./DPDKORG/lib/librte_lpm/Makefile	2013-08-02 11:28:34.486374904 +1200
+++ ./DPDK/lib/librte_lpm/Makefile	2013-07-31 17:08:37.214124693 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_lpm.a
diff -rupN ./DPDKORG/lib/librte_malloc/Makefile ./DPDK/lib/librte_malloc/Makefile
--- ./DPDKORG/lib/librte_malloc/Makefile	2013-08-02 11:28:34.486374904 +1200
+++ ./DPDK/lib/librte_malloc/Makefile	2013-07-31 17:09:03.302150244 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_malloc.a
diff -rupN ./DPDKORG/lib/librte_mbuf/Makefile ./DPDK/lib/librte_mbuf/Makefile
--- ./DPDKORG/lib/librte_mbuf/Makefile	2013-08-02 11:28:34.474374327 +1200
+++ ./DPDK/lib/librte_mbuf/Makefile	2013-07-31 17:09:34.694374965 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_mbuf.a
diff -rupN ./DPDKORG/lib/librte_mempool/Makefile ./DPDK/lib/librte_mempool/Makefile
--- ./DPDKORG/lib/librte_mempool/Makefile	2013-08-02 11:28:34.490127868 +1200
+++ ./DPDK/lib/librte_mempool/Makefile	2013-07-31 17:10:02.055124867 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_mempool.a
diff -rupN ./DPDKORG/lib/librte_net/Makefile ./DPDK/lib/librte_net/Makefile
--- ./DPDKORG/lib/librte_net/Makefile	2013-08-02 11:28:34.474374327 +1200
+++ ./DPDK/lib/librte_net/Makefile	2013-07-31 17:10:26.270289647 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 CFLAGS += $(WERROR_FLAGS) -I$(SRCDIR) -O3
 
diff -rupN ./DPDKORG/lib/librte_pmd_e1000/Makefile ./DPDK/lib/librte_pmd_e1000/Makefile
--- ./DPDKORG/lib/librte_pmd_e1000/Makefile	2013-08-02 11:28:34.470135997 +1200
+++ ./DPDK/lib/librte_pmd_e1000/Makefile	2013-07-31 17:10:58.274124721 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 #
 # library name
diff -rupN ./DPDKORG/lib/librte_pmd_ixgbe/Makefile ./DPDK/lib/librte_pmd_ixgbe/Makefile
--- ./DPDKORG/lib/librte_pmd_ixgbe/Makefile	2013-08-02 11:28:34.490127868 +1200
+++ ./DPDK/lib/librte_pmd_ixgbe/Makefile	2013-07-31 17:11:30.422625509 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 #
 # library name
diff -rupN ./DPDKORG/lib/librte_ring/Makefile ./DPDK/lib/librte_ring/Makefile
--- ./DPDKORG/lib/librte_ring/Makefile	2013-08-02 11:28:34.486374904 +1200
+++ ./DPDK/lib/librte_ring/Makefile	2013-07-31 17:12:04.670176976 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_ring.a
diff -rupN ./DPDKORG/lib/librte_timer/Makefile ./DPDK/lib/librte_timer/Makefile
--- ./DPDKORG/lib/librte_timer/Makefile	2013-08-02 11:28:34.490127868 +1200
+++ ./DPDK/lib/librte_timer/Makefile	2013-07-31 17:12:32.258375260 +1200
@@ -31,6 +31,7 @@
 # 
 
 include $(RTE_SDK)/mk/rte.vars.mk
+include $(RTE_SDK)/mk/rte.shared.mk
 
 # library name
 LIB = librte_timer.a
diff -rupN ./DPDKORG/mk/libtrace.mk ./DPDK/mk/libtrace.mk
--- ./DPDKORG/mk/libtrace.mk	1970-01-01 12:00:00.000000000 +1200
+++ ./DPDK/mk/libtrace.mk	2013-08-02 14:45:01.422143520 +1200
@@ -0,0 +1,6 @@
+# Handles a tiny bit of extra work so that we don't break when doing a
+# DPDK make like what rte.app.mk would do
+
+# Taken from Intel DPDK to put -Wl, before a -melf option otherwise this breaks
+comma := ,
+LDFLAGS := $(addprefix -Wl$(comma),$(LDFLAGS))
diff -rupN ./DPDKORG/mk/rte.shared.mk ./DPDK/mk/rte.shared.mk
--- ./DPDKORG/mk/rte.shared.mk	1970-01-01 12:00:00.000000000 +1200
+++ ./DPDK/mk/rte.shared.mk	2013-08-02 14:53:46.730830978 +1200
@@ -0,0 +1,6 @@
+# Required by libtrace to make Intel DPDK build in such away that it
+# works with a shared library
+
+CFLAGS += -fPIC
+ASFLAGS += -fPIC
+LDFLAGS += -fPIC -shared
